oidc/src/main/java/io/quarkiverse/renarde/oidc/impl/RernardeRevokeController.java

```diff
@@ -0,0 +1,82 @@
+package io.quarkiverse.renarde.oidc.impl;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Base64;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import io.quarkiverse.renarde.Controller;
+import io.quarkiverse.renarde.security.RenardeSecurity;
+import io.quarkus.oidc.AccessTokenCredential;
+import io.quarkus.security.Authenticated;
+import io.smallrye.jwt.algorithm.SignatureAlgorithm;
+import io.smallrye.jwt.build.Jwt;
+
+@Path("_renarde/security")
+public class RernardeRevokeController extends Controller {
+
+    @Inject
+    AccessTokenCredential accessToken;
+
+    @RestClient
+    RenardeAppleClient renardeAppleClient;
+
+    @ConfigProperty(name = "quarkus.oidc.apple.client-id")
+    String appleClientId;
+
+    @ConfigProperty(name = "quarkus.oidc.apple.credentials.jwt.issuer")
+    String appleOidcIssuer;
+
+    @ConfigProperty(name = "quarkus.oidc.apple.credentials.jwt.token-key-id")
+    String appleOidcKeyId;
+
+    @ConfigProperty(name = "quarkus.oidc.apple.credentials.jwt.key-file", defaultValue = "AuthKey_XXX.p8")
+    String appleKeyFile;
+
+    @Inject
+    public RenardeSecurity security;
+
+    @Path("apple-revoke")
+    @Authenticated
+    public Response revokeApple() {
+        String clientSecret = Jwt.audience("https://appleid.apple.com")
+                .subject(appleClientId)
+                .issuer(appleOidcIssuer)
+                .issuedAt(Instant.now().getEpochSecond())
+                .expiresIn(Duration.ofHours(1))
+                .jws()
+                .keyId(appleOidcKeyId)
+                .algorithm(SignatureAlgorithm.ES256)
+                .sign(getPrivateKey(appleKeyFile));
+        renardeAppleClient.revokeAppleUser(appleClientId, clientSecret, accessToken.getToken(), "access_token");
```

Comment:
Perhaps we also need to revoke the refresh token? The docs do mention it.

Reply:
I thought we had choice, using the access token OR the refresh token. I might be wrong. Do you think I should make a double call to this endpoint, one for each ? Starting with the refresh_token ?

Reply:
I'll read again and make some test, if needed, I'll add this second call.

Reply:
Since the refresh token is supposed to be longer-lived than the access token it must be invalidated as well. I just added that part.
I also added a check for the tenant id on the back-end side.